src/map/unit.c

```diff
@@ -1041,11 +1041,20 @@ static int unit_stop_walking(struct block_list *bl, int flag)
 
 static int unit_skilluse_id(struct block_list *src, int target_id, uint16 skill_id, uint16 skill_lv)
 {
-	return unit->skilluse_id2(
-		src, target_id, skill_id, skill_lv,
-		skill->cast_fix(src, skill_id, skill_lv),
-		skill->get_castcancel(skill_id)
-	);
+	int casttime = skill->cast_fix(src, skill_id, skill_lv);
+	int castcancel = skill->get_castcancel(skill_id);
+	int ret = unit->skilluse_id2(src, target_id, skill_id, skill_lv, casttime, castcancel);
+	struct map_session_data *sd = BL_CAST(BL_PC, src);
+
+	if (sd != NULL) {
+		sd->itemskill_id = 0;
```

Comment:
you have these called multiple times, a `pc_itemskill_clear` might help.

Reply:
Good point. I'll outsource this to a separate function.

Reply:
Done.